### PR TITLE
Separate BookRepository and IBook Repository

### PR DESCRIPTION
**Pull request by EwaRayska**
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis?view=aspnetcore-8.0

https://learn.microsoft.com/en-us/aspnet/mvc/overview/older-versions/getting-started-with-ef-5-using-mvc-4/implementing-the-repository-and-unit-of-work-patterns-in-an-asp-net-mvc-application

### CODE REVIEW

**Comment by EwaRayska** [comment]
@oskardudycz Can an you please review this pull request?

File: LibraryApp/Book.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿namespace LibraryApp
+{
+    public class Book
+    {
+        public int Id { get; set; }
+        public string? Title { get; set; }
+        public string? Author { get; set; }
+        public Book() { }
```
Comment (oskardudycz) [inline LibraryApp/Book.cs:8]: **COMMENT** Out of curiosity do you need those two constructors? (public and non-public).

As an additional need you can read also on:
- [primary constructors](https://andrewlock.net/an-introduction-to-primary-constructors-in-csharp-12/),
- [required and init properties](https://antondevtips.com/blog/csharp-init-only-and-required-properties)

You might not need to use them, but at least worth considering.

File: LibraryApp/Program.cs [hunk 2]
```diff
@@ -1,62 +1,68 @@
...
+        new (6, "Tango", "Slawomir Mrozek"),
+        new (7, "Akademia Pana Kleksa", "Jan Brzechwa"),
+        new (8, "Harry Potter", "J.K. Rowling"),
+        new (9, "Hobbit", "J.R.R. Tolkien"),
+        new (10, "Pan Tadeusz", "Adam Mickiewicz"),
+        new (11, "Balladyna", "Juliusz Slowacki")
+    };
+
 // Add services to the container.
+builder.Services.AddSingleton<IEnumerable<Book>>(books);
```
Comment (oskardudycz) [inline LibraryApp/Program.cs:23]: Nice :+1:

File: LibraryApp/BookRepository.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            if(existingBook != null)
+            {
+                existingBook.Title = book.Title;
+                existingBook.Author = book.Author;
+            }
+        }
+
+        public void DeleteBook(int id)
+        {
+            var book = GetBookById(id);
```
Comment (oskardudycz) [inline LibraryApp/BookRepository.cs:30]: **SUGGESTION:** For this case, it doesn't matter, but in general, it's better not to load the whole entity from the database before removing data.

File: LibraryApp.API.Tests/BooksAPITests.cs [hunk 4]
```diff
@@ -1,46 +1,146 @@
...
+            var response = await _client.GetAsync("/books");
 
             //assert
             response.EnsureSuccessStatusCode();
             var stringResult = await response.Content.ReadAsStringAsync();
             var result = await response.Content.ReadFromJsonAsync<Book[]>();
 
             Assert.NotNull(result);
-            Assert.Equal(11, result.Length);
+            Assert.Equal(3, result.Length);
```
Comment (oskardudycz) [inline LibraryApp.API.Tests/BooksAPITests.cs:61]: **CONCERN:** This test is potentially clashing with `Post_AddNewBook` . If you add the book, then it'll increase this default number of books. As a riddle: do you know why this test succeeds?

**Review by oskardudycz — COMMENTED** [review]
@EwaRayska nicely done!

Added a few small comments :+1:

**Pull request merged**